### PR TITLE
user model

### PR DESCRIPTION
**Pull request by SharProd**
create custom user model

### CODE REVIEW

File: Eduplatform_site/account/models.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+from django.db import models
+from django.contrib.auth.models import AbstractUser
+from django.utils.translation import gettext_lazy as _
+from django.utils import timezone
+from .mixins import DateTimeMixinModel
+from .managers import CustomUserManager
+
+
+class User(DateTimeMixinModel,AbstractUser):
```
Comment (nikitapt63) [inline Eduplatform_site/account/models.py]: Why you choose an AbstractUser instead of AbstractBaseUser?

File: Eduplatform_site/account/models.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    last_name = models.CharField(_("last name"), max_length=150, blank=True)
+    email = models.EmailField(_("email address"), unique=True)
+    is_staff = models.BooleanField(_("staff status"),default=False,)
+    is_active = models.BooleanField(_("active"),default=True)
+    date_joined = models.DateTimeField(_("date joined"), default=timezone.now)
+
+    objects = CustomUserManager()
+
+    USERNAME_FIELD = "email"
+    REQUIRED_FIELDS = []
```
Comment (nikitapt63) [inline Eduplatform_site/account/models.py]: Why empty?

File: Eduplatform_site/account/models.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    def __str__(self):
+        return f'{self.id} - {self.email} '
+
+    class Meta:
+        verbose_name = _("Custom_user")
+        verbose_name_plural = _("Castom_users")
```
Comment (nikitapt63) [inline Eduplatform_site/account/models.py]: Just user/users

File: Eduplatform_site/account/tests.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+
+
+class UsersManagersTests(TestCase):
```
Comment (nikitapt63) [inline Eduplatform_site/account/tests.py]: For what testing bulit-in Manager?

File: Eduplatform_site/account/admin.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+class UserAdmin(admin.ModelAdmin):
+    list_display = ('id',
+                    'email',
+                    'is_staff',
+                    )
+    list_display_links = ('id',
+                          'email',
+                          )
+
+
```
Comment (nikitapt63) [inline Eduplatform_site/account/admin.py:11]: Add Teacher and Student models here

File: Eduplatform_site/account/models.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+    class Meta:
+        verbose_name = _("teacher")
+        verbose_name_plural = _("teachers")
+
+
+class Student(models.Model):
+    user_id = models.ForeignKey(User, on_delete=models.CASCADE)
+    photo = models.ImageField(upload_to='photos\ ')
+
```
Comment (nikitapt63) [inline Eduplatform_site/account/models.py:44]: * add completed tests and completed articles fields. 
* add rating field
_______________________
**after creating Test, Course, Article models**

File: Eduplatform_site/account/models.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        return f'{self.id}, {self.user_id}'
+
+    class Meta:
+        verbose_name = _("teacher")
+        verbose_name_plural = _("teachers")
+
+
+class Student(models.Model):
+    user_id = models.ForeignKey(User, on_delete=models.CASCADE)
+    photo = models.ImageField(upload_to='photos\ ')
```
Comment (nikitapt63) [inline Eduplatform_site/account/models.py]: Now its impossible to upload more then one photo.

File: Eduplatform_site/account/models.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    def __str__(self):
+        return f'{self.id}, {self.user_id}'
+
+    class Meta:
+        verbose_name = _("teacher")
+        verbose_name_plural = _("teachers")
+
+
+class Student(models.Model):
+    user_id = models.ForeignKey(User, on_delete=models.CASCADE)
```
Comment (nikitapt63) [inline Eduplatform_site/account/models.py]: just "user"

**Review by nikitapt63 — CHANGES_REQUESTED** [review]

File: Eduplatform_site/account/models.py (hunk 1 above)
Comment (SharProd) [reply Eduplatform_site/account/models.py]: there are more fields to choose from

File: Eduplatform_site/account/models.py (hunk 2 above)
Comment (SharProd) [reply Eduplatform_site/account/models.py]: i edit it, add email and password

File: Eduplatform_site/account/tests.py (hunk 4 above)
Comment (SharProd) [reply Eduplatform_site/account/tests.py]: removed tests

File: Eduplatform_site/account/models.py (hunk 2 above)
Comment (SharProd) [reply Eduplatform_site/account/models.py]: add only password

File: Eduplatform_site/account/admin.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+admin.site.register(User,UserAdmin)
+admin.site.register(Teacher)
+admin.site.register(Student)
+admin.site.register(UserPhotos)
```
Comment (nikitapt63) [inline Eduplatform_site/account/admin.py]: "UserPhotos" model is not necessary

File: Eduplatform_site/account/models.py [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    last_name = models.CharField(_("last name"), max_length=150, blank=True)
+    email = models.EmailField(_("email address"), unique=True)
+    is_staff = models.BooleanField(_("staff status"),default=False,)
+    is_active = models.BooleanField(_("active"),default=True)
+    date_joined = models.DateTimeField(_("date joined"), default=timezone.now)
+
+    objects = CustomUserManager()
+
+    USERNAME_FIELD = "email"
+    REQUIRED_FIELDS = ["password"]
```
Comment (nikitapt63) [inline Eduplatform_site/account/models.py]: "password" doesn't need to be here, you may add "username", "first_name", "last_name" fields here

File: Eduplatform_site/account/models.py [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    def __str__(self):
+        return f'{self.id} - {self.email} '
+
+    class Meta:
+        verbose_name = _("user")
+        verbose_name_plural = _("users")
+
+
+class Teacher(models.Model):
+    user_id = models.ForeignKey(User, on_delete = models.CASCADE)
```
Comment (nikitapt63) [inline Eduplatform_site/account/models.py]: not user_id, just "user"

File: Eduplatform_site/account/models.py (hunk 8 above)
Comment (nikitapt63) [reply Eduplatform_site/account/models.py]: same here

File: Eduplatform_site/account/models.py [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+    class Meta:
+        verbose_name = _("student")
+        verbose_name_plural = _("student")
+
+class UserPhotos(models.Model):
+    photo = models.ImageField(upload_to='image/%Y/%m/%d/')
+    user = models.ForeignKey(User,on_delete=models.CASCADE)
+    teacher = models.ForeignKey(Teacher, on_delete=models.CASCADE, blank=True, null=True)
+    student = models.ForeignKey(Student, on_delete=models.CASCADE, blank=True, null=True)
```
Comment (nikitapt63) [inline Eduplatform_site/account/models.py:55]: Teacher and Student needs single model with image. not in one table, and "use" doesn't need Image it's an abstract entity

**Review by nikitapt63 — CHANGES_REQUESTED** [review]

File: Eduplatform_site/account/admin.py (hunk 9 above)
Comment (nikitapt63) [reply Eduplatform_site/account/admin.py]: in admin.py

File: Eduplatform_site/account/models.py (hunk 12 above)
Comment (nikitapt63) [reply Eduplatform_site/account/models.py:55]: "user"*

**Pull request merged**